File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
from collections import deque
def solution(people, limit):
    answer = 0
    people.sort()
    people = deque(people)
    while len(people) > 1 :
        if people[0] + people[-1] <= limit:
            people.popleft()
            people.pop()
            answer += 1
        else:
            people.pop()
            answer += 1
    if len(people) != 0:
        answer += 1
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    
    let left = 0;
    let right = people.length - 1;
    
    let answer = 0;
    while(left <= right){
        if(people[left] + people[right] <= limit){
            left++;
        }
        answer++;
        right--;
    }
    
    return answer;
}